classes/opensesameapi.php

```diff
@@ -42,106 +42,124 @@
 
-    }
+    /**
+     * @var
+     */
+    protected $courserequesturl;
 
     /**
-     * Get http status code
+     * Constructor.
      *
-     * @return int|boolean status code or false if not available.
+     * @throws \dml_exception
+     * @throws \moodle_exception
      */
-    public function get_http_code() {
-        mtrace('Calling get_http_code.');
-        $info = $this->get_info();
-        if (!isset($info['http_code'])) {
-            return false;
+    public function __construct() {
+        parent::__construct();
+        $this->clientid = get_config('tool_opensesame', 'clientid');
+        $this->clientsecret = get_config('tool_opensesame', 'clientsecret');
+        $this->authurl = get_config('tool_opensesame', 'authurl');
+        $this->baseurl = get_config('tool_opensesame', 'baseurl');
+        $this->customerintegrationid = get_config('tool_opensesame', 'customerintegrationid');
+
+        // Check if each property is set and not null.
+        $properties = ['clientid', 'clientsecret', 'authurl', 'baseurl', 'customerintegrationid'];
+        foreach ($properties as $property) {
+            if (!isset($this->$property) || $this->$property === null) {
+                throw new \moodle_exception($property . '_missing', 'tool_opensesame');
+            }
         }
-        mtrace('returning status code ' . $info['http_code']);
-        return $info['http_code'];
     }
 
     /**
      * Get authenticate  API Credentialing.
      *
-     * @return token if authenticated.
+     * @return bool false if authentication fails false if access_token not set returned
      * @throws \dml_exception
      */
-    public function authenticate(): token {
-        mtrace('Authenticating.');
-        $authurl = get_config('tool_opensesame', 'authurl');
-        $clientid = get_config('tool_opensesame', 'clientid');
-        $clientsecret = get_config('tool_opensesame', 'clientsecret');
-
+    public function authenticate(): bool {
+        mtrace('Authenticating...');
+        $clientid = $this->clientid;
+        $clientsecret = $this->clientsecret;
         $this->setHeader([
                 'Content-Type: application/x-www-form-urlencoded',
                 'Accept: application/json',
                 sprintf('Authorization: Basic %s', base64_encode(sprintf('%s:%s', $clientid, $clientsecret)))
         ]);
-
+        $authurl = $this->authurl;
         $response = $this->post($authurl, 'grant_type=client_credentials&scope=content'
         );
 
         $decoded = json_decode($response);
-        $token = $decoded->access_token;
-        set_config('bearertoken', $token, 'tool_opensesame');
-        set_config('bearertokencreatetime', time(), 'tool_opensesame');
-        $createtime = get_config('tool_opensesame', 'bearertokencreatetime');
-        set_config('bearertokenexpiretime', ($createtime + $decoded->expires_in), 'tool_opensesame');
-        return $this->get_auth_token();
+        if (!$decoded->access_token) {
```

Comment:
Is there an http status code and a message that is returned in the response? It would be helpful to check that status code and if it is not 200, log the code and message.